writer.go

```diff
@@ -134,7 +154,7 @@ func (w *Writer) writeAndRetry(p Priority, s string) (int, error) {
 	w.Lock()
 	defer w.Unlock()
 
-	if w.conn != nil {
+	if w.conn != nil && !w.needReconnect {
```

Comment:
Rather than have a poller in a separate goroutine (which requires more resources and memory protection), I think it'd be better to have a `needsToReconnect()` function that performs the read and if it fails then we can attempt to reconnect before writing. That would solve the problem of losing a log line if the socket has closed, while presenting a lower impact to the process (and the network) overall.